app/react/src/client/preview/render.tsx

```diff
@@ -63,6 +114,29 @@ class ErrorBoundary extends ReactComponent<{
   }
 }
 
+// Will be used to execute a callback function as soon as the React Elements are mounted.
+// This is necessary for the new React Root Api, because passing a callback function to
+// the Root API's render function is not possible anymore.
+class CallbackWrapper extends ReactComponent<{ callback: () => void }, { isDivVisible: boolean }> {
```

Comment:
Are there other docs which show this as a best practice, or is this something that you came up with specifically for this PR? Could you do the same thing with a `useEffect` call?

Reply:
Please take a look at https://github.com/reactwg/react-18/discussions/5 (section "What about the render callback?"). 
Also, https://www.npmjs.com/package/react-callback follows almost the same idea.

Update:

I have transformed the component to be function-based

Reply:
Update:

In my last commit I have removed the CallbackWrapper component and used another approach described here:
https://github.com/reactwg/react-18/discussions/5#discussioncomment-796012

Honestly, I don't know, whether the timing will be perfect with the `setTimeout` method. Unfortunately, we cannot use the `requestIdleCallback` method, because the browser support is not the best (https://caniuse.com/requestidlecallback).

Reply:
Hi @valentinpalkovic! Thanks for your work on this.

As mentioned earlier, I reviewed this behind-the-scenes with @shilman and raised the concern about the `CallbackWraper`. I still have reservations. So I asked in the React Working Group discussion how we could achieve this same callback timing without the complexity of the wrapper.

You can find that [sub-discussion here](https://github.com/reactwg/react-18/discussions/5#discussioncomment-2275882).

In summary, Sebastian Markbåge suggested this approach:

```js
export const CallbackWrapper: React.FC<CallbackProps> = ({
  callback,
  children
}) => {
  let once = useRef(false);
  useLayoutEffect(() = {
    if (once.current) return;
    once.current = true;
    callback();
  }, [callback]);
  return children;
}
```

Reply:
@chantastic the CallbackWrapper was already removed a couple of iterations ago. I am not sure, whether you have reviewed the very latest version of my MR. Also, later on, I have posted some links (https://github.com/storybookjs/storybook/pull/17215#discussion_r805998060) to show up alternative solutions to the mentioned issue of not having a callback function anymore. Please take a look at my follow up comments